__tests__/infrastructure/NodeService.spec.js

```diff
@@ -0,0 +1,229 @@
+					...expectNodeFormattedCommonField,
+					apiEndpoint: 'N/A',
+					roles: 'Peer node',
+					rolesRaw: 1,
+					peerStatus: generateNodePeerStatus(true)
+				},
+				{
+					...nodeCommonField,
+					...expectNodeFormattedCommonField,
+					apiEndpoint: 'localhost.com',
+					roles: 'Peer Api node',
+					rolesRaw: 3,
+					peerStatus: generateNodePeerStatus(true),
+					apiStatus: generateNodeApiStatus(true)
+				},
+				{
+					...nodeCommonField,
+					...expectNodeFormattedCommonField,
+					apiEndpoint: 'N/A',
+					roles: 'Peer Voting node',
+					rolesRaw: 5,
+					peerStatus: generateNodePeerStatus(true)
+				},
+				{
+					...nodeCommonField,
+					...expectNodeFormattedCommonField,
+					apiEndpoint: 'localhost.com',
+					roles: 'Peer Api Voting node',
+					rolesRaw: 7,
+					peerStatus: generateNodePeerStatus(true),
+					apiStatus: generateNodeApiStatus(true)
+				}
+			]);
+		});
+
+		runStatisticServiceFailResponseTests('getNodes', 'getAvailableNodes');
+	});
+
+	describe('getNodeStats', () => {
+		it('return nodes count with 7 types of roles', async () => {
+			// Arrange:
+			http.statisticServiceRestClient = jest.fn().mockImplementation(() => {
+				return {
+					getNodes: jest.fn().mockResolvedValue(statisticServiceNodeResponse)
+				};
+			});
+
+			// Act:
+			const nodeStats = await NodeService.getNodeStats();
+
+			// Assert:
+			expect(nodeStats).toEqual({
+				1: 1,
+				2: 0,
+				3: 1,
+				4: 0,
+				5: 1,
+				6: 0,
+				7: 1
+			});
+		});
+	});
+
+	describe('getNodeInfo', () => {
+		// Arrange:
+		Date.now = jest.fn(() => new Date('2023-02-21'));
+
+		const expectedPeerStatus = {
+			connectionStatus: true,
+			lastStatusCheck: '2023-02-19 12:30:16'
+		};
+
+		const expectedAPIStatus = {
+			apiNodeStatus: true,
+			connectionStatus: false,
+			databaseStatus: true,
+			isHttpsEnabled: true,
+			lastStatusCheck: '2023-02-21 00:00:00',
+			restVersion: '2.4.2'
+		};
+
+		const expectedChainInfoStatus = {
+			height: 2027193,
+			finalizedHeight: 2031992,
+			finalizationEpoch: 1413,
+			finalizationPoint: 7,
+			finalizedHash:
+				'6B687D9B689611C90A1094A7430E78914F22A2570C80D3E42D520EB08091A973',
+			lastStatusCheck: '2023-02-21 00:00:00'
+		};
+
+		const assertNodeStatus = async (node, expectedResult) => {
+			// Arrange:
+			http.statisticServiceRestClient = jest.fn().mockImplementation(() => {
+				return {
+					getNode: jest.fn().mockResolvedValue(node)
+				};
+			});
+
+			// Act:
+			const { apiStatus, chainInfo, peerStatus } =
+				await NodeService.getNodeInfo(node.publicKey);
+
+			// Assert:
+			expect(apiStatus).toEqual(expectedResult.apiStatus);
+			expect(chainInfo).toEqual(expectedResult.chainInfo);
+			expect(peerStatus).toEqual(expectedResult.peerStatus);
+		};
+
+		it('returns api node status and chain info when api status is present', async () => {
+			await assertNodeStatus(statisticServiceNodeResponse[0], {
+				peerStatus: expectedPeerStatus,
+				apiStatus: {},
+				chainInfo: {}
```

Comment:
why is there no chainInfo for peer nodes?

Reply:
The statistic services only provide chain info from the API node.